src/js/setup-text-tracks.js

```diff
@@ -0,0 +1,136 @@
+ *
+ * @private
+ */
+function attachDashTextTracksToVideojs(player, tracks) {
+  const mediaPlayer = player.dash.mediaPlayer;
+
+  const trackDictionaryKeyedByTextTrack = new WeakMap();
+
+  // Add remote tracks
+  tracks
+    // Map input data to match HTMLTrackElement spec
+    // https://developer.mozilla.org/en-US/docs/Web/API/HTMLTrackElement
+    .map((track) => Object.assign(
+      {},
+      track,
+      {
+        default: track.defaultTrack,
+        kind: track.kind,
+        label: track.lang,
+        srclang: track.lang,
+      }
+    ))
+
+    // Add track to videojs track list
+    .map((track) => {
+      const remoteTextTrack = player.addRemoteTextTrack(track, true);
+      trackDictionaryKeyedByTextTrack.set(remoteTextTrack.track, track);
+
+      // Don't add the cues becuase we're going to let dash handle it
+      // natively. This will ensure that dash will stream new text tracks if
+      // necessary because video.js will only render tracks it has stored
+      // locally. Dash is capable of loading separate streams for subtitles
+      //
+      // Example:
+      // https://storage.googleapis.com/shaka-demo-assets/sintel-mp4-wvtt/dash.mpd
+
+      // Return `track` so we can continue chaning.
+      return track;
+    })
+  ;
+
+  /*
+   * Scan `videojs.textTracks()` to find one that is showing. Set the dash
+   * text track.
+   */
+  function updateDashTextTrack() {
+    const dashMediaPlayer = this.dash.mediaPlayer;
+    const textTracks = this.textTracks();
+    let activeTextTrackIndex = -1;
+
+    function findMatchingTrack(dashTrack) {
+      return function({index}) {
+        return index === dashTrack.index;
+      };
+    }
+
+    // Iterate through the tracks and find the one marked as showing.
+    // If none are showing, `activeTextTrackIndex` will be set to
+    // `-1`, disabling text tracks.
+    for (let i = 0; i < textTracks.length; i += 1) {
+      const textTrack = textTracks[i];
+      if (textTrack.mode === 'showing') {
+        const dashTrackToActivate = trackDictionaryKeyedByTextTrack.get(textTrack);
+        const dashTextTrackMediaInfos = dashMediaPlayer.getTracksFor('text');
+        const matchingTrack = dashTextTrackMediaInfos.find(findMatchingTrack(dashTrackToActivate));
+        const matchingTrackIndex = dashTextTrackMediaInfos.indexOf(matchingTrack);
+
+        activeTextTrackIndex = matchingTrackIndex;
+      }
+    }
+
+    // If the text track has changed, then set it in dash
+    if (activeTextTrackIndex !== dashMediaPlayer.getCurrentTextTrackIndex()) {
+      dashMediaPlayer.setTextTrack(activeTextTrackIndex);
+    }
+  }
+
+  // Initialize the text track
+  mediaPlayer.updateDashTextTrack = updateDashTextTrack.bind(player);
+  mediaPlayer.updateDashTextTrack();
+
+  // Update dash when videojs's selected text track changes. Use an arrow function so
+  // `mediaPlayer.updateDashTextTrack` can be overriden exterally.
```

Comment:
This was a mistake, I had used this logic elsewhere and forgot to remove it. I'll remove it here.